codec-http2/src/main/java/io/netty/handler/codec/http2/ReadOnlyHttp2Headers.java

```diff
@@ -823,12 +823,14 @@ private void calculateNext() {
             for (; i < current.length; i += 2) {
                 AsciiString roName = current[i];
                 if (roName.hashCode() == nameHash && roName.contentEqualsIgnoreCase(name)) {
-                    next = current[i + 1];
-                    i += 2;
+                    if (i + 1 < current.length) {
+                        next = current[i + 1];
+                        i += 2;
+                    }
                     return;
                 }
             }
-            if (i >= current.length && current == pseudoHeaders) {
+            if (current == pseudoHeaders) {
```

Comment:
revert... I think this still needs to remain here.

Reply:
LGTM and IDE gives warnings for always `true`.

Reply:
https://lgtm.com/projects/g/netty/netty/snapshot/a2994f5265f7bd7c4bf206d95b643a3c68d4f3c3/files/codec-http2/src/main/java/io/netty/handler/codec/http2/ReadOnlyHttp2Headers.java?sort=name&dir=ASC&mode=heatmap#xe38ef04bb58e4:1

![Screenshot (22)](https://user-images.githubusercontent.com/24762260/95719312-4277c680-0c8d-11eb-87fa-459ef79a23b2.png)


What to do?

Reply:
fine then...